### PR TITLE
NetInfo: Added “connectionChange" event  since “change” event is deprecated.

### PR DESCRIPTION
**Pull request by piyush965**
**Issue description**: 
The “change” event is deprecated and warns when used. “connectionChange" event implementation is missing in both UWP and WPF platforms.

**Expected behaviour**:
 I expect “connectionChange" event should work like as in iOS and Android.

**Proposed Solution** 
Added “connectionChange" support on both platforms (UWP and WPF). This provides access to ConnectionType and EffectiveConnectionType to determine connection level as well as connection type.  
Note: The updated code does not impact existing “change” event interfaces.

**Following test cases are executed:**

**Sample Code:**
NetInfo.addEventListener(
  'connectionChange',
  handleFirstConnectivityChange
);

function handleFirstConnectivityChange(connectionInfo) {
  console.log('type: ' + connectionInfo.type + ', effectiveType: ' + connectionInfo.effectiveType);
}

**ON UWP app:**

**Connected to WIFI**
 log:  type: “InternetAccess”, effectiveType: “unknown”

**Disconnected from WIFI**
log: type: “none”, effectiveType: “unknown”

**Connected to LAN**
 log:  type: “InternetAccess”, effectiveType: “unknown”

**Disconnected from LAN**
log: type: “none”, effectiveType: “unknown”

**Connected to Mobile Data (LTE) - Tested on Windows Phone**
 log:  type: “InternetAccess”, effectiveType: “4g”

**Disconnected from Mobile Data (LTE)**
log: type: “none”, effectiveType: “unknown”

**ON WPF app:**

**Connected to WIFI**
 log:  type: “InternetAccess”, effectiveType: “unknown”

**Disconnected from WIFI**
log: type: “none”, effectiveType: “unknown”

**Connected to LAN**
 log:  type: “InternetAccess”, effectiveType: “unknown”

**Disconnected from LAN**
log: type: “none”, effectiveType: “unknown”

**Connected to Mobile Data (LTE)**
 Not applicable

**Disconnected from Mobile Data (LTE)**
Not applicable

I have submitted a PR with a proposed solution. Please review.

### CODE REVIEW

**Comment by piyush965** [comment]
Related to https://github.com/Microsoft/react-native-windows/issues/1670

**Comment by piyush965** [comment]
@rozele @matthargett Can you please review the changes and provide inputs. Thanks.

**Comment by rozele** [comment]
Nice!

File: ReactWindows/ReactNative/Modules/NetInfo/DefaultNetworkInformation.cs [hunk 1]
```diff
@@ -48,6 +59,38 @@ public NetworkConnectivityLevel ConnectivityLevel
                     return _profile.GetNetworkConnectivityLevel();
                 }
             }
+
+            public string ConnectivityType
+            {
+                get
+                {
+                   WwanDataClass effectiveConnnectivityType = _profile.IsWwanConnectionProfile ? _profile.WwanConnectionProfileDetails.GetCurrentDataClass() : WwanDataClass.None;
```
Comment (matthargett) [inline ReactWindows/ReactNative/Modules/NetInfo/DefaultNetworkInformation.cs]: use var keyword

File: ReactWindows/ReactNative.Net46/Modules/NetInfo/DefaultNetworkInformation.cs [hunk 2]
```diff
@@ -11,6 +11,10 @@ class DefaultNetworkInformation : INetworkInformation
         public event EventHandler<NetworkConnectivityChangedEventArgs> NetworkConnectivityChanged;
         private NetworkListManager _networkListManager;
```
[the remark this answers is not in the document]
Comment (piyush965) [reply ReactWindows/ReactNative.Net46/Modules/NetInfo/DefaultNetworkInformation.cs]: If I make _networkList field an INetworkListManager, the NetworkConnectivityChanged event is not defined on INetworkListManager interface due to this getting compilation errors.

File: ReactWindows/ReactNative.Net46/Modules/NetInfo/DefaultNetworkInformation.cs [hunk 3]
```diff
@@ -11,6 +11,10 @@ class DefaultNetworkInformation : INetworkInformation
...
 
+        private string CONNECTION_TYPE_NONE = "none";
+
+        private string CONNECTION_TYPE_NONE_DEPRECATED = "None";
+
         public void Start()
         {
             _networkListManager = new NetworkListManager();
```
[the remark this answers is not in the document]
Comment (piyush965) [reply ReactWindows/ReactNative.Net46/Modules/NetInfo/DefaultNetworkInformation.cs]: Nothing is passed to DefaultNetworkInformation constructor from NetInfoModule class. Can you please help me to fix this?

**Pull request closed without merging**